src/boot/efi/pe.c

```diff
@@ -153,15 +155,30 @@ static void locate_sections(
         assert(offsets);
         assert(sizes);
 
+        EFI_STATUS err;
+
         for (size_t i = 0; i < n_table; i++) {
                 const PeSectionHeader *sect = section_table + i;
 
                 for (size_t j = 0; sections[j]; j++) {
                         if (memcmp(sect->Name, sections[j], strlen8(sections[j])) != 0)
                                 continue;
 
-                        offsets[j] = in_memory ? sect->VirtualAddress : sect->PointerToRawData;
-                        sizes[j] = sect->VirtualSize;
+                        size_t offset = in_memory ? sect->VirtualAddress : sect->PointerToRawData;
+                        size_t size = sect->VirtualSize;
+
+                        sizes[j] = offsets[j] ? : size;
```

Comment:
Shouldn't this be `sizes[j] = sizes[j] ?: size;`?
Not only that, but this whole logic is written in a rather confusing and 'unique' fashion.

Reply:
I had that previously but changed it due to the comment [1]. However in the version I'm working on this is no longer an issue since this logic cannot be here as we need to find all .dtb sections to correctly compute digests.

[1]: https://github.com/systemd/systemd/pull/28959#discussion_r1319636703